pkg/sysfsnet/interfaces_linux.go

```diff
@@ -23,7 +23,23 @@ func Interfaces() ([]Interface, error) {
 	ifaces := make([]Interface, 0, 16)
 
 	readMACFromFile := func(s string) (string, error) {
+
+		// Check if parent exists and is a directory
+		parent := filepath.Dir(s)
+		info, err := os.Stat(parent)
+		if err != nil {
+			if os.IsNotExist(err) {
+				return "", nil
+			}
+			return "", err
+		}
+
+		if !info.IsDir() {
+			return "", nil
+		}
```

Comment:
What if we do this check earlier, and call `readMACFromFile()` only if it's a dir?

Will this work?
```diff
diff --git a/pkg/sysfsnet/interfaces_linux.go b/pkg/sysfsnet/interfaces_linux.go
index f4a295f..138b788 100644
--- a/pkg/sysfsnet/interfaces_linux.go
+++ b/pkg/sysfsnet/interfaces_linux.go
@@ -39,6 +39,9 @@ func Interfaces() ([]Interface, error) {
 	}
 
 	for _, dentry := range dents {
+		if !dentry.IsDir() {
+			continue
+		}
 		hwText, err := readMACFromFile(filepath.Join(sysClassNet, dentry.Name(), "address"))
 		if os.IsNotExist(err) {
 			continue
```

Reply:
I agree this is much cleaner - I'll make the change and test locally

Reply:
Thanks - looks like we are very close to merging this. Let me know if you need anything.

Reply:
Looks good now with that change, I did have to call os.Stat() first becuase os.DirEntry(dentry) resolves to a symlink and IsDir()/FileInfo doesn't check the target